### ISSUE TITLE
LoggingAwareExecuter implementation reason.

### ISSUE THREAD

**Issue by Furriz844** [xenit-eu/dynamic-extensions-for-alfresco#278]
## I'm submitting a ...   (check one with "x")
```
[] bug report
[ ] feature request
[x ] question
```
Why com.github.dynamicextensionsalfresco.actions.AnnotationBasedActionExecuter doesn't implement LoggingAwareExecuter?

I have exception:
```
Exception in thread "defaultAsyncAction19" java.lang.ClassCastException: com.github.dynamicextensionsalfresco.actions.AnnotationBasedActionExecuter cannot be cast to org.alfresco.repo.action.executer.LoggingAwareExecuter
        at org.alfresco.repo.action.ActionServiceImpl.onLogException(ActionServiceImpl.java:1830)
        at org.alfresco.repo.action.AsynchronousActionExecutionQueueImpl$ActionExecutionWrapper.run(AsynchronousActionExecutionQueueImpl.java:449)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by kerkhofsd** [comment]
Hi @Furriz844 ,

Thanks for pointing this out. 

Looks like Alfresco introduced some new code / components since Alfresco 6 to refactor the reporting of exceptions in the `AsynchronousActionExecutionQueueImpl`:
- `LoggingAwareExecuter`
- `RuntimeActionService`

The reason that the `AnnotationBasedActionExecuter` doesn't implement the `LoggingAwareExecuter` interface, is simply because this interface didn't exist in Alfresco 5.2. 
We'll need to have a more thorough look at this, investigating a solution that is compatible with both Alfresco 5 and 6. 

If I understand correctly, this issue (`ClassCastException`) only pops up when an exception is thrown during the execution of the Action?

**Comment by kerkhofsd** [comment]
Small correction: looks like the `LoggingAwareExecuter` is already around since Alfresco 5.1.

**Comment by Furriz844** [comment]
> 
> 
> Hi @Furriz844 ,
> 
> Thanks for pointing this out.
> 
> Looks like Alfresco introduced some new code / components since Alfresco 6 to refactor the reporting of exceptions in the `AsynchronousActionExecutionQueueImpl`:
> 
>     * `LoggingAwareExecuter`
> 
>     * `RuntimeActionService#onLogException(...)`
> 
> 
> The reason that the `AnnotationBasedActionExecuter` doesn't implement the `LoggingAwareExecuter` interface, is simply because this interface didn't exist in Alfresco 5.2.
> We'll need to have a more thorough look at this, investigating a solution that is compatible with both Alfresco 5 and 6.
> 
> If I understand correctly, this issue (`ClassCastException`) only pops up when an exception is thrown during the execution of the Action?

Thank you for answer!



> 
> 
> Small correction: looks like the `LoggingAwareExecuter` is already around since Alfresco 5.1.

Yes, i am researching it now, because i have alfresco 5.1 and i can see the interface in jar. 

Can we implement LoggingAwareExecuter?

**Comment by kerkhofsd** [comment]
@Furriz844 

The straight forward solution: make `AnnotationBasedActionExecuter` implement the `LoggingAwareExecuter` interface would mean we drop support for Alfresco 5.0. As long as we support Alfresco 5.0, we'll need to do more investigation on a solution that works on all supported Alfresco versions.

**Comment by kerkhofsd** [comment]
Fixed by #284
